packages/core/src/browser/fetchObservable.ts

```diff
@@ -93,25 +94,58 @@ function afterSend(
   responsePromise: Promise<Response>,
   startContext: FetchStartContext
 ) {
-  const reportFetch = (response: Response | Error) => {
-    const context = startContext as unknown as FetchCompleteContext
-    context.state = 'complete'
-    context.duration = elapsed(context.startClocks.timeStamp, timeStampNow())
-
+  const reportFetch = (context: FetchCompleteContext, response: Response | Error) => {
     if ('stack' in response || response instanceof Error) {
       context.status = 0
       context.isAborted = response instanceof DOMException && response.code === DOMException.ABORT_ERR
       context.error = response
-
-      observable.notify(context)
     } else if ('status' in response) {
       context.response = response
       context.responseType = response.type
       context.status = response.status
       context.isAborted = false
+    }
+    observable.notify(context)
+  }
 
-      observable.notify(context)
+  const onFulfilled = (response: Response) => {
+    const context = startContext as unknown as FetchCompleteContext
+    context.state = 'complete'
+    context.resolveDuration = elapsed(context.startClocks.timeStamp, timeStampNow())
+
+    const responseClone = response.clone()
+    const reader = responseClone.body?.getReader()
+
+    if (reader && ReadableStream) {
+      new ReadableStream({
+        start(controller) {
+          return pump()
+
+          function pump(): Promise<undefined> {
+            return (reader as ReadableStreamDefaultReader<Uint8Array>).read().then(({ done }) => {
```

Comment:
💬 **suggestion**: You should handle the case where `.read()` returns a rejected promise (for example, when the request has been cancelled via an `AbortController`). Also `monitor()` those callbacks.

Reply:
we already use `monitor` on `responsePromise.then(monitor(onFulfilled), monitor(onRejected))`, does it make sense to have it here too?

Reply:
Ah that's right the promise is handled somewhere else. Nevermind my comment then!